libs/coin-modules/coin-bitcoin/src/buildTransaction.ts

```diff
@@ -41,9 +43,25 @@ export const buildTransaction = async (
   const walletAccount = getWalletAccount(account);
   const utxoPickingStrategy = selectUtxoPickingStrategy(walletAccount, transaction.utxoStrategy);
 
+  const floorSatPerVB = await getRelayFeeFloorSatVb(walletAccount.xpub.explorer);
+
+  // --- Clamp user/endpoint fee to ≥ floor + 1 sat/vB, return integer sat/vB ---
+  const originalFeeBN = feePerByte;
+  const safeFeeBN = BigNumber.max(originalFeeBN, floorSatPerVB.plus(1)).integerValue(
+    BigNumber.ROUND_CEIL,
+  );
```

Comment:
This would be equivalent to call the clamp function I suggested factoring out after !